javaslang-benchmark/src/test/java/javaslang/JmhRunner.java

```diff
@@ -71,9 +72,28 @@ private static void runAndReport(Class<?> benchmarkClass, int warmupIterations,
         }
     }
 
-    public static <T> void assertEquals(T a, T b) {
-        if (!Objects.equals(a, b)) {
-            throw new IllegalStateException(a + " != " + b);
+    public static <T> void require(T value, Predicate<T> predicate) {
+        require(() -> predicate.test(value));
+    }
+
+    public static <T1, T2> void require(T1 value1, Predicate<T1> predicate1, T2 value2, Predicate<T2> predicate2) {
+        require(() -> predicate1.test(value1),
+                () -> predicate2.test(value2));
+    }
+
+    public static <T1, T2, T3> void require(T1 value1, Predicate<T1> predicate1, T2 value2, Predicate<T2> predicate2, T3 value3, Predicate<T3> predicate3) {
+        require(() -> predicate1.test(value1),
+                () -> predicate2.test(value2),
+                () -> predicate3.test(value3));
+    }
+
+    public static void require(BooleanSupplier... suppliers) {
```

Comment:
I also have few reservations about this `require` call. I personally prefer `assert`

What is the overhead introduced by creating the lambda + making the call in very quick O(1) complexity operations?  If, let's say, the achieve 10M operations/second, we will also be creating 10M lambdas + making 10M calls to `require`.  I know that the overhead with `asserts` will be zero, but I cannot say the same about `require`.